sympy/ntheory/ecm.py

```diff
@@ -341,7 +341,7 @@ def ecm(n, B1=10000, B2=100000, max_curve=200, seed=1234):
                 n //= prime
     while(n > 1):
         try:
-            factor = _ecm_one_factor(n, B1, B2, max_curve, seed + n)
+            factor = _ecm_one_factor(n, B1, B2, max_curve, seed)
```

Comment:
I saw this on the first review and thought it made sense, but now I think this underscores the wrongness of it. Could you look at the intro to sympy/core/random.py and see if you could convert this to use randint and seed as described there?

Reply:
Is it something like this?

```python
from sympy.core.random import randint, seed as seed_func

def _ecm_one_factor(n, B1=10000, B2=100000, max_curve=200, seed=None):
    seed_func(seed)
    ...
    sigma = randint(6, n - 1)
```

This way, the behavior is different when `seed=None`. However, it is not an important difference in `ecm`.

Do you want to fix the other places that use `_randint` as well? If so, we will need to consider them individually.